core/src/test/java/kafka/testkit/TestKitNodes.java

```diff
@@ -97,14 +96,32 @@ public Builder setBrokerNodes(int numBrokerNodes, int disksPerBroker) {
                 if (!brokerNodeBuilders.isEmpty()) {
                     nextId = brokerNodeBuilders.lastKey() + 1;
                 }
-                BrokerNode.Builder brokerNodeBuilder = new BrokerNode.Builder()
+                BrokerNode.Builder brokerNodeBuilder = BrokerNode.builder()
                         .setId(nextId)
                         .setNumLogDirectories(disksPerBroker);
                 brokerNodeBuilders.put(nextId, brokerNodeBuilder);
             }
             return this;
         }
 
+        /**
+         * Set per broker properties overrides, this setter must be invoked after setBrokerNodes which
```

Comment:
The happen-before is anti-pattern in builder pattern. Could we merge `PerBrokerPropertiesOverrides` to `setNumControllerNodes/setNumBrokerNodes`? for example:

```java
setBrokerNodes(int numBrokerNodes, int disksPerBroker, Function<Integer, Map<String, Object>> perBrokerConfigs)

```

Reply:
It seems `TestKitNodes` need to be refactor also. In short, it should verify/apply all settings when building.